### PR TITLE
Migrated documentation from Jupyter to Documenter.jl

### PR DESCRIPTION
**Pull request by dwijenchawra**
Migrated docs to Documenter.jl and updated github deployment workflows.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,6 @@
 
 This library supports representation, inference, and learning in Bayesian networks.
 
-Please read the [documentation](http://nbviewer.ipython.org/github/sisl/BayesNets.jl/blob/master/doc/BayesNets.ipynb).
+Please read the [documentation](https://dwijenchawra.github.io/BayesNets.jl/dev/index.html).
```
Comment (mykelk) [inline README.md:5]: This needs to be updated.

**Review by mykelk — CHANGES_REQUESTED** [review]
Minor change requested.

File: README.md (hunk 1 above)
Comment (dwijenchawra) [reply README.md]: Fixed.

**Comment by dwijenchawra** [comment]
There is a test for documentation that looks for the old Jupyter notebook, is that still necessary? The old notebook was moved to the docs folder for easy access, should it remain there or be deleted?

**Comment by mykelk** [comment]
Cool. I'm good to merge. I was thinking it would be good to have @tawheeler do a quick skim before merging.

**Comment by mykelk** [comment]
Maybe keep the notebook there for testing purposes. It helps with code coverage. We can move the tests later on.

**Comment by tawheeler** [comment]
Moving to Documenter.jl is a really good idea. I like that images can be generated from the markdown. Thanks!

**Comment by mykelk** [comment]
It looks like there is a failure here according to travis:
```
LoadError: SystemError: opening file "/home/travis/build/sisl/BayesNets.jl/doc/BayesNets.ipynb": No such file or directory
```
Can we make sure that file is added back? Once that's done, let's get this merged!

**Comment by dwijenchawra** [comment]
There was a refactor needed in the tests files, we changed the name of the doc folder to docs because thats the way it was structured in both FreudenthalTriangulation and the Documenter docs. Both builds have passed now so I think we are good to go as well!

**Pull request merged**